tests/thread_test.c

```diff
@@ -38,9 +37,8 @@ static int s_test_thread_creation_join_fn(struct aws_allocator *allocator, void
     ASSERT_INT_EQUALS(
         AWS_THREAD_JOINABLE, aws_thread_get_detach_state(&thread), "thread state should have returned JOINABLE");
     ASSERT_SUCCESS(aws_thread_join(&thread), "thread join failed");
-    ASSERT_INT_EQUALS(
-        test_data.thread_id,
-        aws_thread_get_id(&thread),
+    ASSERT_TRUE(
```

Comment:
Maybe have an assert false test here too?

Reply:
To do that, we would need to manufacture an aws_thread_id value that we know is not the current thread. One way would be to create a new thread and then check against that.